src/Cli/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -54,133 +75,123 @@ public int Execute()
             try
             {
                 ICommand targetCommand = GetTargetCommand();
-                if (launchSettings != null)
-                {
-                    if (!string.IsNullOrEmpty(launchSettings.ApplicationUrl))
-                    {
-                        targetCommand.EnvironmentVariable("ASPNETCORE_URLS", launchSettings.ApplicationUrl);
-                    }
-
-                    targetCommand.EnvironmentVariable("DOTNET_LAUNCH_PROFILE", launchSettings.LaunchProfileName);
-
-                    foreach (var entry in launchSettings.EnvironmentVariables)
-                    {
-                        string value = Environment.ExpandEnvironmentVariables(entry.Value);
-                        //NOTE: MSBuild variables are not expanded like they are in VS
-                        targetCommand.EnvironmentVariable(entry.Key, value);
-                    }
-                    if (string.IsNullOrEmpty(targetCommand.CommandArgs) && launchSettings.CommandLineArgs != null)
-                    {
-                        targetCommand.SetCommandArgs(launchSettings.CommandLineArgs);
-                    }
-                }
-
+                var launchSettingsCommand = ApplyLaunchSettingsProfileToCommand(targetCommand, launchSettings);
```

Comment:
Extracted all the guts of the launch profile application to a separate method for readability.